pkg/controller/bookkeepercluster/bookkeepercluster_controller.go

```diff
@@ -233,35 +234,51 @@ func (r *ReconcileBookkeeperCluster) syncBookieSize(bk *bookkeeperv1alpha1.Bookk
 
 func (r *ReconcileBookkeeperCluster) reconcileFinalizers(bk *bookkeeperv1alpha1.BookkeeperCluster) (err error) {
 	if bk.DeletionTimestamp.IsZero() {
-		if !util.ContainsString(bk.ObjectMeta.Finalizers, util.ZkFinalizer) {
-			bk.ObjectMeta.Finalizers = append(bk.ObjectMeta.Finalizers, util.ZkFinalizer)
+		// checks whether the slice of finalizers contains a string with the given prefix
+		// NOTE: we need to ensure that no two finalizer names have the same prefix
+		if !util.ContainsStringWithPrefix(bk.ObjectMeta.Finalizers, util.ZkFinalizer) {
+			finalizer := util.ZkFinalizer
+			configMap := &corev1.ConfigMap{}
+			if strings.TrimSpace(bk.Spec.EnvVars) != "" {
+				err = r.client.Get(context.TODO(), types.NamespacedName{Name: strings.TrimSpace(bk.Spec.EnvVars), Namespace: bk.Namespace}, configMap)
+				if err == nil {
```

Comment:
What if err !=nil. This will be the case if user has not set the "bk.Spec.EnvVars" field.
We need to change this check to :
if err != nil {
return err
}
/// rest of the code here for err == nil